x-pack/plugins/core_telemetry/index.js

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { registerCollectors } from './server/lib/collectors';
+import { registerTasks, scheduleTasks } from './server/lib/tasks';
+
+export const coreTelemetry = (kibana) => {
```

Comment:
This PR creates a new X-Pack plugin, `core_telemetry`. I considered some alternatives before deciding to add this:

 - Can it be in `src/legacy/core_plugins/kibana/server/lib`, next to the KQL usage collector?
   Since this logic depends on the X-Pack plugin `task_manager`, I think it makes the most sense for it to be a new X-Pack plugin. It doesn't seem possible for some code in core kibana to depend on services exposed by an X-Pack plugin.
 - Can it be in `x-pack/plugins/xpack_main`? 
  It's possible, but it would require making `xpack_main` "depend" on `task_manager`. In that case, `task_manager` could never depend on `xpack_main` for license checks, because it would create a circular dependency.

Having a plugin for for this one piece of functionality seems a little bit overspecific, but I think it will be simple to clean up a little by moving the KQL usage collector into this plugin. This new plugin creates a home for other future "core" usage collectors to live in.